collector/connectors/odigosrouterconnector/routingmap.go

```diff
@@ -0,0 +1,101 @@
+package odigosrouterconnector
+
+import (
+	"fmt"
+
+	"github.com/odigos-io/odigos/common"
+	"github.com/odigos-io/odigos/common/pipelinegen"
+)
+
+// RoutingIndex maps signals (logs/metrics/traces) to dataStream pipelines
+type RoutingIndex map[common.ObservabilitySignal][]string // signal -> []pipeline names
+
+// SignalRoutingMap indexes all sources and namespaces by SourceKey and provides routing per signal
+//
+//	 SignalRoutingMap{
+//	    {ns1/deployment/frontend}: {
+//	        "logs":    ["dataStream-A"],
+//	        "traces":  ["dataStream-A", "dataStream-B"],
+//	        "metrics": ["dataStream-B"],
+//	    },
+//	    {ns2/statefulset/db}: {
+//	        "traces": ["dataStream-B"],
+//	    },
+//	    {ns3/*/*}: {
+//	        "traces": ["dataStream-Default"],
+//	    },
+//	}
+type SignalRoutingMap map[string]RoutingIndex
+
+// BuildSignalRoutingMap prepares a fast-access routing map based on structured group details.
+// Future-proof: usable by both routing connector and custom connector logic.
+func BuildSignalRoutingMap(dataStreams []pipelinegen.DataStreams) SignalRoutingMap {
+	result := make(SignalRoutingMap)
+
+	for _, dataStream := range dataStreams {
+
+		signalsForDataStream := GetSignalsForDataStream(dataStream)
+
+		// Build the keys for the sources
+		for _, source := range dataStream.Sources {
+			key := fmt.Sprintf("%s/%s/%s", source.Namespace, NormalizeKind(source.Kind), source.Name)
+
+			if _, exists := result[key]; !exists {
+				result[key] = make(RoutingIndex)
+			}
+
+			for _, signal := range signalsForDataStream {
+				pipeline := dataStream.Name
+				result[key][signal] = appendIfMissing(result[key][signal], pipeline)
+			}
+		}
+	}
+
+	return result
+}
+
+// normalizeKind ensures kind comparisons are case-insensitive and aligned with OTel semantic keys
+// e.g: Deployment -> deployment, StatefulSet -> statefulset, DaemonSet -> daemonset
+func NormalizeKind(kind string) string {
+	switch kind {
+	case "Deployment", "deployment":
+		return "deployment"
+	case "StatefulSet", "statefulset":
+		return "statefulset"
+	case "DaemonSet", "daemonset":
+		return "daemonset"
+	default:
+		return kind
+	}
+}
```

Comment:
can this be simplified to `strings.ToLower`?
Also, this one is missing CronJob.